integration-test/helpers/harness.js

```diff
@@ -279,7 +279,18 @@ export async function printPerformanceSummary (name, times) {
  * @param {string[]} [names]
  * @returns {Promise<MockCall[]>}
  */
-export async function mockedCalls (page, names = []) {
+export async function mockedCalls (page, names = [], mustExist = true) {
+    if (names.length > 0 && mustExist) {
+        await page.waitForFunction(({names}) => {
+            const calls = window.__playwright.mocks.calls
+            return calls.some(([name]) => names.includes(name))
+        }, {names})
+    }
+
+    if (!mustExist) {
+        await page.waitForTimeout(500)
+    }
```

Comment:
We discussed this over MM. The `mustExist` flag is used to ensure that Playwright waits until the function appears (or timeout). On the other hand, if the flag is false we wait to make sure it's not a false negative.